transformer/Models.py

```diff
@@ -24,22 +24,27 @@ class PositionalEncoding(nn.Module):
 
     def __init__(self, d_hid, n_position=200):
         super(PositionalEncoding, self).__init__()
-
+        self.dims = torch.arange(0, d_hid)
+        self.positions = torch.arange(n_position) 
+        self.dims.requires_grad = False
+        self.positions.requires_grad = False
+        # codes above are to support `torch.Tensor.to` or `torch.Tensor.cuda` operation
         # Not a parameter
-        self.register_buffer('pos_table', self._get_sinusoid_encoding_table(n_position, d_hid))
+        self.register_buffer('pos_table', self._get_sinusoid_encoding_table())
 
-    def _get_sinusoid_encoding_table(self, n_position, d_hid):
+    def _get_sinusoid_encoding_table(self):
         ''' Sinusoid position encoding table '''
-        # TODO: make it with torch instead of numpy
-
-        def get_position_angle_vec(position):
-            return [position / np.power(10000, 2 * (hid_j // 2) / d_hid) for hid_j in range(d_hid)]
 
-        sinusoid_table = np.array([get_position_angle_vec(pos_i) for pos_i in range(n_position)])
-        sinusoid_table[:, 0::2] = np.sin(sinusoid_table[:, 0::2])  # dim 2i
-        sinusoid_table[:, 1::2] = np.cos(sinusoid_table[:, 1::2])  # dim 2i+1
+        def get_position_angle_vec():
+            dims = 2 * (self.dims.long() // 2) / d_hid # shape: dims,
+            divided = torch.pow(10000, dims.float()) # shape: dims,
+            return self.positions / divided.contiguous().view(divided.size(0), *[1] * self.positions.dim()) # shape: *positions.size(), dims
+            
+        sinusoid_table = get_position_angle_vec()
+        sinusoid_table[:, 0::2] = sinusoid_table[:, 0::2].sin()  # dim 2i
+        sinusoid_table[:, 1::2] = sinusoid_table[:, 0::2].cos()  # dim 2i+1
```

Comment:
hi, I think this should be "sinusoid_table[:, 1::2] = sinusoid_table[:, 1::2].cos()"